### ISSUE TITLE
Provide a way to get `resume_offset` of a swap file

### ISSUE THREAD

**Issue by nl6720** [util-linux/util-linux#2724]
As mentioned in https://github.com/util-linux/util-linux/issues/2359, btrfs-progs has a [`btrfs inspect-internal map-swapfile`](https://man.archlinux.org/man/btrfs.5#HIBERNATION) command to get a usable `resume_offset` value for hibernation purposes.

Getting `resume_offset` [on any other swapfile-supporting file system](https://wiki.archlinux.org/title/Power_management/Suspend_and_hibernate#Hibernation_into_swap_file) requires using `filefrag` from e2fsprogs and parsing its output.

It would be great if `resume_offset` could be acquired with some tool from `util-linux` alone.

**Comment by endrift** [comment]
I wonder if it would make sense to have this be a separate tool from mkswap since you could conceivably want to get that on an already existing swap file, not just a new one. Adding a flag to mkswap that makes it _not_ make the swap file but return info instead might be confusing.

**Comment by karelzak** [comment]
The thing seems complex. It is not about resume_offset only. I would expect a tool that will be able to set
```
/sys/power/resume
/sys/power/resume_offset
```
for regular swap as for swapfile, and unfortunately, it also depends on the used filesystem (in case of btrfs), LVM,  etc.

It would be nice to have an overview of what is already implemented (for example, in systemd).

**Comment by nl6720** [comment]
I just want a tool for getting the value, not one that automatically populates `/sys/power/resume*`.

**Comment by YHNdnzj** [comment]
In systemd we have https://github.com/systemd/systemd/blob/main/src/shared/hibernate-util.c#L198 for this purpose.

Note that since the implementation of `btrfs inspect-internal map-swapfile` is not exposed in libbtrfs, I manually duplicated their code in https://github.com/systemd/systemd/pull/29179. If there shall be more users, I think it's more graceful to ask btrfs-progs people to add that, instead of having a third reimplementation...

**Comment by karelzak** [comment]
Yes, that's why I'd like to see an overview of the current situation.

**Comment by nl6720** [comment]
Slightly related: [mm: swap: print starting physical block offset in swapon](https://lore.kernel.org/all/20240522074658.2420468-3-Sukrit.Bhatnagar%40sony.com/). If this lands, the kernel will print the offset to dmesg.
